libs/sdk-common/src/input_parser.rs

```diff
@@ -183,8 +191,21 @@ use crate::prelude::*;
 pub async fn parse(
     input: &str,
     external_input_parsers: Option<&[ExternalInputParser]>,
+    dns_resolver: Option<&AsyncResolver<GenericConnector<TokioRuntimeProvider>>>,
```

Comment:
Any reason to have this as an optional parameter? It won't be exposed to the bindings anyway so why not use our own inside the function and simplify the interface?

Reply:
I agree with you. I think that I misunderstood this comment to the liquid PR (https://github.com/breez/breez-sdk-liquid/pull/619/files/51cf0e48605d5df88f26a3b12a21e2332c89e607)

Reply:
See https://github.com/breez/breez-sdk-greenlight/pull/1149#discussion_r1894510251